### PR TITLE
Update FDT interface

### PR DESCRIPTION
**Pull request by jameslu8**
- Add PayloadBase node
 - Deprecate PCI attribute and UID/HID
 - Mark extra-data as optional
 - Deprecate  cpu-info->iospace

### CODE REVIEW

File: source/2_universal_payload.rst [hunk 1]
```diff
@@ -2834,13 +2928,90 @@ The below tables show what nodes should be passed from bootloader and its corres
...
+| smbios               | entry                | <base address>          | Point to smbios structure UNIVERSAL_PAYLOAD_SMBIOS_TABLE                                        |
++----------------------+----------------------+-------------------------+-------------------------------------------------------------------------------------------------+
+
++----------------------+----------------------+-------------------------+-------------------------------------------------------------------------------------------------+
+| Node                 | Property             | Value                   | Definition                                                                                      |
++======================+======================+=========================+=================================================================================================+
+| PayloadBase          | entry                | <base address>          | Point to ELF image base on permenant memory                                                     |
```
Comment (ChaselChiu) [inline source/2_universal_payload.rst:3010]: on permanent memory or flash.

Comment (jameslu8) [reply source/2_universal_payload.rst:3010]: I assume ELF on flash is compressed hence not able to be parsed?

Comment (ChaselChiu) [reply source/2_universal_payload.rst:3010]: yes, I would like to support both cases.
if ELF is compressed, this base address will be permanent memory.
if ELF is not compressed, this base address will be flash.

**Review by ChaselChiu — APPROVED** [review]

**Review by gdong1 — APPROVED** [review]
looks good to me.

**Pull request merged**